### ISSUE TITLE
Annual average price for electricity

### ISSUE THREAD

**Issue by nairnamratha** [quintel/etsource#1607]
How is the red, dotted line for _annual average price_ for electricity obtained? I can't seem to find how this was calculated. 

I would like to use this annual average price for electricity to calculate the total costs of electricity used as a "transport fuel". However, simply doing AVG (MERIT_PRICE_CURVE()) doesn't work because MERIT_PRICE_CURVE() returns no values for the current year.

![screen shot 2018-03-13 at 15 10 03](https://user-images.githubusercontent.com/32448293/37346841-bd18441e-26d0-11e8-8dbc-07f01bede97e.png)
@antw @jorisberkhout @ChaelKruip @michieldenhaan

**Comment by antw** [comment]
>  However, simply doing AVG(MERIT_PRICE_CURVE()) doesn't work because MERIT_PRICE_CURVE() returns no values for the current year.

The merit order only runs for the `future` year and therefore the price curve will be empty in the `present`.

You can get the average cost with `Q(average_costs_of_produced_electricity)` which references a gquery: https://github.com/quintel/etsource/blob/5a636f49b5084f2da4a95a601a202a311e26ef71/gqueries/general/costs/average_costs_of_produced_electricity.gql#L1-L8

**Comment by nairnamratha** [comment]
I had checked that gquery, but was confused because the value that the gquery gives is not the same as the annual average price value shown on the graph. 

The graph shows a value somewhere around _33 EUR/MWh_, but the gquery gives a result around _0.0185 euro/mJ_ which is about _66 EUR/MWh_ when converted. 

So, my questions are: 

1.  Why these values are different?
2.  Why is the gquery value about twice the value shown on the graph?
3.  Which one is the correct result for "annual average electricity price" ?

**Comment by nairnamratha** [comment]
Also, I talked to @grdw and with his help, I decided to use:
"future:AVG(MERIT_PRICE_CURVE())/SECS_PER_HOUR" as a part of my query to represent the annual average electricity price. 

Does this make sense ? Or is it "illegal" to go about it by only considering the future prices?

**Comment by antw** [comment]
`MERIT_PRICE_CURVE()` describes the marginal cost of producing 1 MWh with the most expensive running producer.  The "average_costs_of_produced_electricity" query is an average of the *total* costs of generating each MWh of electricity (including initial investment, depreciation, and fixed operation and maintenance costs).

I think a rough equivalent to `AVG(MERIT_PRICE_CURVE())` using the graph only would be:

```ruby
DIVIDE( 
  V(G(electricity_production), "marginal_costs * output_of_electricity").sum, 
  V(G(electricity_production), output_of_electricity).sum 
)
```

Annoyingly, this graph version is 5-10 times slower than simply averaging the merit price curve.

Unfortunately using `MERIT_PRICE_CURVE()` will also not work if the user opts to disable the merit order. Here's a query which will use `MERIT_PRICE_CURVE()` when the merit order is enabled, and fall back to the above graph approach when it is disabled:

```ruby
IF(
  GRAPH(use_merit_order_demands?),
  -> {
    AVG(MERIT_PRICE_CURVE())
  },
  -> {
    DIVIDE( 
      V(G(electricity_production), "marginal_costs * output_of_electricity").sum, 
      V(G(electricity_production), output_of_electricity).sum 
    )
  }
)
```

**Comment by nairnamratha** [comment]
@antw Thanks! That makes sense and it's working now

**Comment by ChaelKruip** [comment]
> Why these values are different?

The query/queries takes into account all costs of electricity plants:
```
total_costs = fixed_costs + variable_costs
```
where the fixed costs include investment and fixed O&M costs etc. 

The E-price curve (the thing that Merit produces) uses `marginal_costs` to determine a proxy for the electricity price. This marginal cost basically takes only the variable part of the total costs into account. This result is necessarily lower than the total with the exception that the Merit module assigns a very special HIGH price to moments where there is insufficient installed capacity which could drive the curve up to something like 600 EUR/MWh if I'm not mistaken.

In short: 
* The Merit price curve shows the marginal cost (EUR/MWh) of the 'most expensive' running plant. This price can be such that power plants are not earning their investments back. 
* The query takes all costs of electricity production into account (including investments and other fixed costs)

**Comment by marliekeverweij** [comment]
@ChaelKruip can this issue be closed?

**Comment by ChaelKruip** [comment]
> @ChaelKruip can this issue be closed?

Yes. Closing
